### PR TITLE
Add/config flags storage update

### PR DESCRIPTION
**Pull request by gmovr**
Related to Automattic/martech#1509

## Proposed Changes

* Adding a feature flag for storage add-on. E.g. for conditional visibility, testing, and to guard this feature until ready.
* Adding a selector to check the flag. For now it's just checking the config flag, but we will extend this with other conditions, e.g. certain plans if needed, and it makes it easier in case the feature flag gets renamed, or the logic changes otherwise. I didn't add any tests for it, because it's just taking a boolean from the config object and returns it, and casting it to boolean via `!!` (since it's reading from an untyped JSON file). 

## Testing Instructions

* Check that all the environments have this flag, and that it's set to `false`.
* For manual testing, you can also import the selector anywhere you'd like, e.g. the plans page, and `console.log` it, or add a `debugger;` statement inside the selector and verify that the value is read correctly (in this case it should return `false`).

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Resolves Automattic/martech#1509

### CODE REVIEW

File: client/state/selectors/is-storage-addon-available.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import config from '@automattic/calypso-config';
+
+/**
+ * Is the storage addon (space upgrade) available, based on config flag vs. environment.
+ *
+ * @returns {boolean} - Whether or not the storage addon is available
+ */
+const isStorageAddonAvailable = (): boolean => !! config.isEnabled( 'storage-addon' );
```
Comment (chriskmnds) [inline client/state/selectors/is-storage-addon-available.ts]: This probably shouldn't live under `state/selectors` if it's not interacting with state directly :-)

Comment (gmovr) [reply client/state/selectors/is-storage-addon-available.ts]: Cool, that makes sense. I was kind of envisioning the config as global state, but technically it isn't. Will move it into the addons folder👍

Comment (chriskmnds) [reply client/state/selectors/is-storage-addon-available.ts]: Yeah. The selectors here don't have any connection to state, not outside of what is passed into them. So unless you are passing state into the exported function, then it probably doesn't belong here. I don't _think_ there are exceptions.

**Comment by ddc22** [comment]
Probably stale and can be closed off @gmovr @jeyip ?

**Comment by jeyip** [comment]
> Probably stale and can be closed off @gmovr @jeyip ?

After reading the PR description, I believe so 👍. Closing this -- we can always reopen if I'm incorrect @gmovr

**Pull request closed without merging**